web/html/src/utils/functions.ts

```diff
@@ -119,14 +119,14 @@ function capitalize(str: string): string {
 
 function generatePassword(): string {
   const length = Math.floor(Math.random() * 10) + 15;
+  // See https://stackoverflow.com/a/68617567/1470607
   const charset = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789:-_";
+  const array = new Uint32Array(charset.length);
+  window.crypto.getRandomValues(array);
+
   let password = "";
-  if (window.crypto && window.crypto.getRandomValues) {
-    var rand = new Uint16Array(length);
-    window.crypto.getRandomValues(rand);
-    for (let i = 0; i < length; i++) password += charset.charAt(Math.floor((rand[i] * charset.length) / 65536));
-  } else {
-    for (let i = 0; i < length; i++) password += charset.charAt(Math.floor(Math.random() * charset.length));
+  for (let i = 0; i < length; i++) {
+    password += charset[array[i] % charset.length];
```

Comment:
Is `array[i] % charset.length` always an integer without using `Math.floor`?
Or is there some typescript magic that does that automatically?

Reply:
We used to index by `Math.random()` in `Math.floor(Math.random() * charset.length)` which would need the floor, now we index by `i` which is from the loop body and will be `0`, `1`, `2` etc. The difference is that now `array` is already full of random values (due to `window.crypto.getRandomValues(array)`) so we can just index into it one element after the other and get random values out.

Reply:
I was talking about the index in charset[]. As it looked to me it's not always going to be an integer if we divide by `charset.length`